src/Cake.Tfs/PullRequest/TfsPullRequestSettings.cs

```diff
@@ -66,6 +66,44 @@ public TfsPullRequestSettings(TfsPullRequestSettings settings)
             this.ThrowExceptionIfPullRequestCouldNotBeFound = settings.ThrowExceptionIfPullRequestCouldNotBeFound;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TfsPullRequestSettings"/> class using environment variables.
+        /// </summary>
+        /// <param name="credentials">Credentials to use to authenticate against Team Foundation Server or
+        /// Azure DevOps.</param>
+        public TfsPullRequestSettings(ITfsCredentials credentials)
+        {
+            credentials.NotNull(nameof(credentials));
+            this.Credentials = credentials;
+
+            var repositoryUrl = Environment.GetEnvironmentVariable("BUILD_REPOSITORY_URI", EnvironmentVariableTarget.Process);
+            repositoryUrl.NotNullOrWhiteSpace(nameof(repositoryUrl));
+            this.RepositoryUrl = new Uri(repositoryUrl);
+
+            var pullRequestId = Environment.GetEnvironmentVariable("SYSTEM_PULLREQUEST_PULLREQUESTID", EnvironmentVariableTarget.Process);
+            pullRequestId.NotNullOrWhiteSpace(nameof(pullRequestId));
+            if (!int.TryParse(pullRequestId, out int pullRequestIdValue))
+            {
+                throw new ArgumentException("SYSTEM_PULLREQUEST_PULLREQUESTID should contain integer value", nameof(pullRequestId));
+            }
+
+            pullRequestIdValue.NotNegativeOrZero(nameof(pullRequestId));
+            this.PullRequestId = pullRequestIdValue;
+        }
+
+        /// <summary>
+        /// Constructs the settings object using the provided access token.
+        /// </summary>
+        /// <returns>The instance of <see cref="TfsPullRequestSettings"/> class.</returns>
+        public static TfsPullRequestSettings UsingTfsBuildOAuthToken()
+        {
+            var accessToken = Environment.GetEnvironmentVariable("SYSTEM_ACCESSTOKEN", EnvironmentVariableTarget.Process);
+            accessToken.NotNullOrWhiteSpace(nameof(accessToken));
```

Comment:
This will throw an `ArgumentNullException` which might be misleading as no argument is involved. It would be better to throw an `InvalidOperation` exception. It would also be helpful to point the user in the right direction as probably it's neither not running in an Azure Pipelines build or the build definition doesn't have the `Allow scripts to access the OAuth token` option not set.